src/components/panel/panel.spec.js

```diff
@@ -1053,6 +1053,114 @@ describe('$mdPanel', function() {
     });
   });
 
+  describe('grouping logic:', function() {
+    it('should create a group using the newPanelGroup method', function() {
+      $mdPanel.newPanelGroup('test');
+
+      expect($mdPanel._groups['test']).toExist();
+    });
+
+    it('should create a group using the config option groupName when the ' +
+        'group hasn\'t been created yet', function() {
+          var config = {
+            groupName: 'test'
+          };
+          var panel = $mdPanel.create(config);
+
+          expect($mdPanel._groups['test']).toExist();
+        });
+
+    it('should add a panel to a group using the addToGroup method', function() {
+      $mdPanel.newPanelGroup('test');
+      var panel = $mdPanel.create(DEFAULT_CONFIG);
+
+      panel.addToGroup('test');
+      expect(getGroupPanels('test')).toContain(panel);
+    });
+
+    it('should add a panel to a group using the config option groupName',
+        function() {
+          $mdPanel.newPanelGroup('test');
+
+          var config = {
+            groupName: 'test'
+          };
+
+          var panel = $mdPanel.create(config);
+          expect(getGroupPanels('test')).toContain(panel);
+        });
+
+    it('should remove a panel from a group using the removeFromGroup method',
```

Comment:
Missing tests:

```
it('should remove a panel from a group on panel destroy', ...
it('should update open panels when a panel is closed', ... // Similar to the closing the first panel when you hit maxOpen,
// but you manually close one and make sure the final state is still correct (that another one isn't also closed).
it('should throw if trying to set maxOpen on a group that doesn/'t exist', ...
it('should only create a group once', ... // Create a group using config, then create a group with newPanelGroup.

// For a future PR because I don't think you have this functionality yet...
it('should reopen a forced close panel when another is closed', ...
```